plugins/SegmentEditor/javascripts/Segmentation.js

```diff
@@ -1278,28 +1278,7 @@ $(document).ready(function() {
             segmentDefinition = cleanupSegmentDefinition(segmentDefinition);
             segmentDefinition = encodeURIComponent(segmentDefinition);
 
-            if (piwikHelper.isAngularRenderingThePage()) {
-
-                angular.element(document).injector().invoke(function ($location, $rootScope) {
-                    var $search = $location.search();
-
-                    if (segmentDefinition !== $search.segment) {
-                        // eg when using back button the date might be actually already changed in the URL and we do not
-                        // want to change the URL again
-                        $search.segment = segmentDefinition.replace(/%$/, '%25').replace(/%([^\d].)/g, "%25$1");
-                        $location.search($search);
-                        setTimeout(function () {
-                            try {
-                                $rootScope.$apply();
-                            } catch (e) {}
-                        }, 1);
-                    }
-
-                });
-                return false;
-            } else {
-                return broadcast.propagateNewPage('segment=' + segmentDefinition, true);
-            }
+            return broadcast.propagateNewPage('segment=' + segmentDefinition, true);
```

Comment:
It is a big downside that when adding/updating segments the page will reload. But we can't afford to introduce other bugs in this release. maybe we can patch the editor to work well without reload in 3.1 or so (without doing full rewrite which we can't do at this time).

Good thing is that changing segments still load fast..